dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/ExternalSourceRestRepository.java

```diff
@@ -89,10 +89,10 @@ public ExternalSourceRest findOne(Context context, String externalSourceName) {
     }
 
     @Override
+    @PreAuthorize("permitAll()")
     public Page<ExternalSourceRest> findAll(Context context, Pageable pageable) {
         List<ExternalDataProvider> externalSources = externalDataService.getExternalDataProviders();
-        return converter.toRestPage(externalSources, pageable, externalSources.size(),
```

Comment:
why is the total dropped?

Reply:
the original code was buggy as the method toRestPage with 4 parameters expect that the list contains only the requested page. When the pagination is not directly supported by the service it is needed to use the 3 parameters form that will automatically extract the right subset from the whole list. There were no test failure as the endpoint was only tested with a single mock external source provider. Please note that this bug and in general the implementation of this endpoint is not part of our PR that was scoped to add support for live import provider